light/provider/http/http.go

```diff
@@ -149,6 +177,11 @@ func (p *http) validatorSet(ctx context.Context, height *int64) (*types.Validato
 
 			case *url.Error:
 				if e.Timeout() {
+          // if we have exceeded retry attempts then return a no response error 
+          if attempt == p.maxRetryAttempts {
```

Comment:
```suggestion
          if attempt >= p.maxRetryAttempts {
```

Reply:
We can do this, but I don't think it's possible for attempt to ever be greater than `maxRetryAttempts` because we always terminate when they are equal and `maxRetryAttempts` is a uint (i.e. can't be negative)